controllers/agents/svc/serviceclass_controller.go

```diff
@@ -102,7 +103,14 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
-	if err = r.setOwnerReference(ctx, &serviceClass, req.Namespace); err != nil {
+	// get the controller's deployment
+	controller := appsv1.Deployment{}
+	controllerRef := types.NamespacedName{Namespace: serviceClass.Namespace, Name: constants.ServiceAgentDeploymentName}
+	if err = r.Get(ctx, controllerRef, &controller); err != nil {
+		reconcileLog.Error(err, "Failed to retrieve controller reference")
+	}
```

Comment:
I'd say we should handle the error here

```suggestion
	if err := r.Get(ctx, controllerRef, &controller); err != nil {
		reconcileLog.Error(err, "Failed to retrieve controller reference")
		if apierrors.IsNotFound(err) {
			panic(err)
		}
		return ctrl.Result{}, err
	}
```

Reply:
I'm not sure the panic is enough, since controller-runtime installs a panic handler around the reconciler.  In this case, should we call `os.Exit(1)`?

Reply:
I'd go for `os.Exit(1)`. This way we are protecting the application from accidentally being installed with a different name. WDYT?

Reply:
I'm not a huge fan of shutting down the entire pod; that seems like a great way to cause problems.  If there was a cleaner way of asking the controller to self-terminate, I'd be all for that.

Reply:
I agree, we can find a better way for that. I'd propose to go with `os.Exit(1)` for now, add a `TODO` comment here, and open an issue for taking track.